### PR TITLE
[REF] base_setup: convert to new js framework

### PR DESCRIPTION
**Pull request by SimonGenin**
entreprise Pr: https://github.com/odoo/enterprise/pull/29804

### CODE REVIEW

File: addons/base_setup/static/src/js/res_config_invite_users.js [hunk 1]
```diff
@@ -1,174 +1,116 @@
...
-                    $userEmails.prop('disabled', false);
-                    this.$('.o_web_settings_invite').prop('disabled', false);
-                    self.$('.o_user_emails').focus();
-                }
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { _t } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+import { unique } from "@web/core/utils/arrays";
```
Comment (aab-odoo) [inline addons/base_setup/static/src/js/res_config_invite_users.js:6]: import order

File: addons/base_setup/static/src/js/res_config_invite_users.js [hunk 2]
```diff
@@ -1,174 +1,116 @@
...
+class ResConfigInviteUsers extends Component {
+    setup() {
+        this.orm = useService("orm");
+        this.invite = useService("user_invite");
+        this.action = useService("action");
+        this.notification = useService("notification");
+
+        this.state = useState({
+            status: "idle", // idle, inviting
+            emails: "",
```
Comment (aab-odoo) [inline addons/base_setup/static/src/js/res_config_invite_users.js:19]: default invite

File: addons/base_setup/static/src/js/res_config_invite_users.js [hunk 3]
```diff
@@ -1,174 +1,116 @@
...
+     * @param {string} email
+     * @returns {boolean} true if the given email address is valid
+     */
+    validateEmail(email) {
+        var re = /^([a-z0-9][-a-z0-9_\+\.]*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,63}(?:\.[a-z]{2})?)$/i;
+        return re.test(email);
+    }
+
+    get emails() {
+        const emails = unique(this.state.emails.split(/[ ,;\n]+/));
```
Comment (aab-odoo) [inline addons/base_setup/static/src/js/res_config_invite_users.js:37]: t-model.lazy

File: addons/base_setup/static/src/js/res_config_invite_users.js [hunk 4]
```diff
@@ -1,174 +1,116 @@
...
+        return emails;
+    }
+
+    validate() {
+        if (!this.emails.length) {
+            throw new Error(_t("Empty email address"));
+        }
+        for (const email of this.emails) {
+            if (!this.validateEmail(email)) {
+                throw new Error(`${_t("Invalid email address")}: ${email}`);
```
Comment (aab-odoo) [inline addons/base_setup/static/src/js/res_config_invite_users.js:50]: throw one error for all invalid adresses

File: addons/base_setup/static/src/js/res_config_invite_users.js [hunk 5]
```diff
@@ -1,174 +1,116 @@
...
+            this.state.invite = await this.invite.data();
+        })
+    }
+
+    /**
+     * @param {string} email
+     * @returns {boolean} true if the given email address is valid
+     */
+    validateEmail(email) {
+        var re = /^([a-z0-9][-a-z0-9_\+\.]*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,63}(?:\.[a-z]{2})?)$/i;
```
Comment (aab-odoo) [inline addons/base_setup/static/src/js/res_config_invite_users.js:32]: const (severa times)

File: addons/base_setup/static/src/js/res_config_invite_users.js [hunk 6]
```diff
@@ -1,174 +1,116 @@
...
+        const emails = unique(this.state.emails.split(/[ ,;\n]+/));
+        if (emails.length === 1 && !emails[0]) {
+            return [];
+        }
+        return emails;
+    }
+
+    validate() {
+        if (!this.emails.length) {
+            throw new Error(_t("Empty email address"));
```
Comment (aab-odoo) [inline addons/base_setup/static/src/js/res_config_invite_users.js:46]: Tab on empty input should not be prevent defaulted, and do nothing here

File: addons/base_setup/static/src/js/res_config_edition.js [hunk 7]
```diff
@@ -1,26 +1,24 @@
...
-                : moment().add(30, 'd');
-        },
-   });
+/**
+ * Widget in the settings that handles a part of the "About" section.
+ * Contains info about the odoo version, database expiration date and copyrights.
+ */
+class ResConfigEdition extends Component {
+    setup() {
+        this.server_version = session.server_version;
```
Comment (aab-odoo) [inline addons/base_setup/static/src/js/res_config_edition.js:15]: camelcase

Comment (aab-odoo) [reply addons/base_setup/static/src/js/res_config_edition.js:15]: whtie list base_setup in tooling

File: addons/base_setup/static/src/xml/res_config_dev_tool.xml [hunk 8]
```diff
@@ -1,16 +1,16 @@
...
-                        <a t-if="!widget.isAssets" class="d-block" href="?debug=assets">Activate the developer mode (with assets)</a>
-                        <a t-if="!widget.isTests" class="d-block" href="?debug=assets,tests">Activate the developer mode (with tests assets)</a>
-                        <a t-if="widget.isDebug" class="d-block" href="?debug=">Deactivate the developer mode</a>
-                        <a t-if="widget.isDebug and !widget.demo_active" class="o_web_settings_force_demo" href="#">Load demo data</a>
+                        <a t-if="!isDebug" class="d-block" href="?debug=1">Activate the developer mode</a>
+                        <a t-if="!isAssets" class="d-block" href="?debug=assets">Activate the developer mode (with assets)</a>
+                        <a t-if="!isTests" class="d-block" href="?debug=assets,tests">Activate the developer mode (with tests assets)</a>
+                        <a t-if="isDebug" class="d-block" href="?debug=">Deactivate the developer mode</a>
+                        <a t-if="isDebug and !demo.isDemoDataActive()" t-on-click.prevent="() => this.onClickForceDemo()" class="o_web_settings_force_demo" href="#">Load demo data</a>
```
Comment (aab-odoo) [inline addons/base_setup/static/src/xml/res_config_dev_tool.xml:13]: this.isDemoDataActive = await this.demo.isDemoDataActive(); in willStart

Comment (aab-odoo) [reply addons/base_setup/static/src/xml/res_config_dev_tool.xml:13]: t-on-click.prevent="() => this.onClickForceDemo()" -> t-on-click.prevent="onClickForceDemo"

File: addons/base_setup/static/src/xml/res_config_edition.xml [hunk 9]
```diff
@@ -1,13 +1,11 @@
...
-<!-- TODO Also see what is done to override the (Community Edition) -->
 <template>
-    <div t-name='res_config_edition'>
+    <div t-name='res_config_edition' owl="1">
         <div class="col-12 o_setting_box" id="edition">
             <div class="o_setting_right_pane">
                 <div class="user-heading">
                     <h3 class="px-0">
-                        Odoo <t t-esc="widget.server_version"/>
+                        Odoo <t t-esc="server_version"/>
```
Comment (aab-odoo) [inline addons/base_setup/static/src/xml/res_config_edition.xml:8]: camelcase

File: addons/base_setup/static/src/xml/res_config_invite_users.xml [hunk 10]
```diff
@@ -1,20 +1,20 @@
 <?xml version="1.0" encoding="utf-8"?>
 <templates>
-    <div t-name='res_config_invite_users'>
+    <div t-name='res_config_invite_users' owl="1">
         <p class="o_form_label">Invite New Users</p>
+
         <div class="d-flex">
-            <input class="o_user_emails o_input mt8" type="text" placeholder="Enter e-mail address"/>
-            <button class="btn btn-primary o_web_settings_invite" data-loading-text="Inviting..."><strong>Invite</strong></button>
+            <input t-model="state.emails" t-att-disabled="state.status != 'idle'" class="o_user_emails o_input mt8" t-on-keydown="(ev) => this._onKeydownUserEmails(ev)" type="text" placeholder="Enter e-mail address"/>
```
Comment (aab-odoo) [inline addons/base_setup/static/src/xml/res_config_invite_users.xml:7]: t-on-keydown="(ev) => this._onKeydownUserEmails(ev)" -> t-on-keydown="_onKeydownUserEmails"

Comment (aab-odoo) [reply addons/base_setup/static/src/xml/res_config_invite_users.xml:7]: several handlers can be "simplified" like this

**Review by aab-odoo — COMMENTED** [review]
type "abc@ozeofezk.com ," doesn't work

**Comment by SimonGenin** [comment]
@aab-odoo fixed

**Review by aab-odoo — APPROVED** [review]
robodoo rebase-ff r+

**Pull request closed without merging**